WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -435,4 +454,108 @@ - (void)defaultAccountDidChange:(NSNotification *)notification
+    if (count == 0) {
+        self.notificationBadgeIconView.hidden = YES;
+        return;
+    }
+
+    BOOL wasNotificationBadgeHidden = self.notificationBadgeIconView.hidden;
+    self.notificationBadgeIconView.hidden = NO;
+    if (wasNotificationBadgeHidden) {
+        [self animateNotificationBadgeIcon];
+    }
+}
+
+#pragma mark - NSObject(NSKeyValueObserving) Helpers
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
+{
+    if ([keyPath isEqualToString:NSStringFromSelector(@selector(applicationIconBadgeNumber))]) {
+        [self updateNotificationBadgeVisibility];
+    }
+}
+
+#pragma mark - Notification Badge Icon Management
+
+- (void)addNotificationBadgeIcon
+{
+    self.notificationBadgeIconView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, WPNotificationBadgeIconSize, WPNotificationBadgeIconSize)];
+    self.notificationBadgeIconView.image = [UIImage imageWithColor:[WPStyleGuide jazzyOrange]
+                                                        havingSize:CGSizeMake(WPNotificationBadgeIconSize, WPNotificationBadgeIconSize)];
+    self.notificationBadgeIconView.layer.cornerRadius = WPNotificationBadgeIconSize / 2.0;
+    self.notificationBadgeIconView.layer.masksToBounds = YES;
+    self.notificationBadgeIconView.layer.borderColor = [[UIColor whiteColor] CGColor];
+    self.notificationBadgeIconView.layer.borderWidth = 1.0;
+    self.notificationBadgeIconView.hidden = YES;
+    [self.view addSubview:self.notificationBadgeIconView];
+
+    [self updateNotificationBadgeIconPosition];
+    [self updateNotificationBadgeVisibility];
+}
+
+- (void)updateNotificationBadgeIconPosition
+{
+    UIInterfaceOrientation orientation = [[UIApplication sharedApplication] statusBarOrientation];
+    BOOL isLandscape = (orientation == UIInterfaceOrientationLandscapeLeft) || (orientation == UIInterfaceOrientationLandscapeRight);
+
+    // We need to take the extra space before & after the tabbar into account for iPad and iPhone 6 Plus
+    CGFloat horizontalOffset = 0.0;
+    if (IS_IPAD) {
+        horizontalOffset = isLandscape ? WPNotificationBadgeIconHorizontalOffsetForIPadInLandscape : WPNotificationBadgeIconHorizontalOffsetForIPadInPortrait;
+    }
+    else if (isLandscape && WPDeviceIdentification.isiPhoneSixPlus) {
+        horizontalOffset = WPNotificationBadgeIconHorizontalOffsetForIPhone6PlusInLandscape;
+    }
+    CGFloat verticalPosition = CGRectGetHeight(self.view.bounds) - (self.tabBar.frame.size.height - WPNotificationBadgeIconVerticalOffsetFromTop);
+    // Subtract the space before & after the tabbar
+    CGFloat tabBarContentWidth = self.tabBar.frame.size.width - (horizontalOffset * 2);
+    CGFloat tabItemWidth = tabBarContentWidth / self.tabBar.items.count;
+    CGFloat horizontalPosition = ((WPTabNotifications + 0.5) * tabItemWidth) - WPNotificationBadgeIconHorizontalOffsetFromCenter + horizontalOffset;
```

Comment:
It's just 1/2, it's not a magical number, hence there is no name to it. It's used to get the center position of notification tab item. Maybe I can separate the `(WPTabNotifications + 0.5) * tabItemWidth)` part to a different variable such as `notificationTabBarItemCenterPosition` if it's not clear. What do you think?